check_diff/src/lib.rs

```diff
@@ -1,11 +1,59 @@
 use std::env;
 use std::io;
 use std::path::Path;
+use std::path::PathBuf;
 use std::process::Command;
+use std::process::Output;
 use tracing::info;
 
+pub enum CheckDiffError {
+    FailedGit(GitError),
+    FailedCommand(String),
+    FailedUtf8(String),
+    FailedSourceBuild(String),
+    FailedCopy(String),
+    FailedCargoVersion(String),
+    FailedVersioning(String),
+    FailedPathBuf(String),
+    IO(std::io::Error),
+}
+
+pub struct GlobalRunners {
```

Comment:
How do you feel about introducing this `GlobalRunners` struct? The purpose of this struct is to allow other functions to have access to both source and feature runners while ensuring easy access to either one.